features/steps/s4s.py

```diff
@@ -115,20 +116,41 @@ def step_impl(context, mu_ccds_query):
     query = mu_ccds_query.format(patientId=context.vendor_config['versioned_api'].get('patient'))
     response = utils.get_resource(context, query)
 
+    context.is_fallback = False
+
+    # we want to support an older way to retrieve smoking status, using the
+    # social-history parameter instead of the code; if the code-based method
+    # fails, fall back to the social-history method
+    if mu_ccds_query == MU_CCDS_MAPPINGS['Smoking status'] and not response.ok:
+        fallback_query = MU_CCDS_MAPPINGS['Smoking status fallback'].format(
+            patientId=context.vendor_config['versioned_api'].get('patient')
+        )
+        response = utils.get_resource(context, fallback_query)
+        context.primary_uri = query
+        context.fallback_uri = fallback_query
+        context.is_fallback = True
+
     context.response = response
 
 
 @given('I have a {mu_ccds_query:MU_CCDS} response')
-def step_impl(context, mu_ccds_query):
-    query = mu_ccds_query.format(patientId=context.vendor_config['versioned_api'].get('patient'))
-
+def step_impl(context, mu_ccds_query):  # pylint: disable=W0613
     assert context.response is not None, \
         'Missing response.'
 
     context.execute_steps('then the response code should be 200')
 
-    assert query in context.response.request.url, \
-        'Missing {query}'.format(query=query)
+    assert not context.response.history, \
```

Comment:
I think the original code here would trigger the failure only during redirects where the redirected URL doesn't contain the initially requested path+query string. I needed to change it for the fallback logic so in its place we can just check the request history, which is nonempty if there have been redirects. It's not exactly the same logic - the new way triggers a failure even if the initially requested path+query are in the redirected URI - but should the server ever be redirecting to some other base URI? Is that allowed in the workflow?